instrumentation/mysql2/test/opentelemetry/instrumentation/mysql2/instrumentation_test.rb

```diff
@@ -309,6 +309,136 @@
           end
         end
       end
+
+      describe 'when span_name is set as statement_type' do
+        it 'sets span name to statement type' do
+          OpenTelemetry::TestHelpers.with_env('OTEL_RUBY_INSTRUMENTATION_MYSQL2_CONFIG_OPTS' => 'span_name=statement_type') do
+            instrumentation.instance_variable_set(:@installed, false)
+            instrumentation.install
+
+            sql = "SELECT * from users where users.id = 1 and users.email = 'test@test.com'"
+            expect do
+              client.query(sql)
+            end.must_raise Mysql2::Error
+
+            _(span.name).must_equal 'select'
+          end
+        end
+
+        it 'sets span name to mysql when statement type is not recognized' do
+          OpenTelemetry::TestHelpers.with_env('OTEL_RUBY_INSTRUMENTATION_MYSQL2_CONFIG_OPTS' => 'span_name=statement_type') do
+            instrumentation.instance_variable_set(:@installed, false)
+            instrumentation.install
+
+            sql = 'DESELECT 1'
+            expect do
+              client.query(sql)
+            end.must_raise Mysql2::Error
+
+            _(span.name).must_equal 'mysql'
+          end
+        end
+      end
+
+      describe 'when span_name is set as db_name' do
+        it 'sets span name to db name' do
+          OpenTelemetry::TestHelpers.with_env('OTEL_RUBY_INSTRUMENTATION_MYSQL2_CONFIG_OPTS' => 'span_name=db_name') do
+            instrumentation.instance_variable_set(:@installed, false)
+            instrumentation.install
+
+            sql = "SELECT * from users where users.id = 1 and users.email = 'test@test.com'"
+            expect do
+              client.query(sql)
+            end.must_raise Mysql2::Error
+
+            _(span.name).must_equal 'mysql' # TODO: change the db name so we can distinguish it from the default
```

Comment:
Could also stub out the `database_name` method in the test, but that is less good than what's proposed in this TODO 🤷